### ISSUE TITLE
Plugin Development plugin declares inputs with custom equals()

### ISSUE THREAD

**Issue by DanielThomas** [gradle/gradle#20033]
```
> Task :nebula-code-quality:pluginDescriptors
Using objects as inputs that have a different serialized form but are equal has been deprecated. This is scheduled to be removed in Gradle 8.0. Type 'org.gradle.plugin.devel.PluginDeclaration' has a custom implementation for equals(). Declare the property as @Nested instead to expose its properties as inputs. See https://docs.gradle.org/7.4/userguide/upgrading_version_7.html#equals_up_to_date_deprecation for more details.
```

### Your Environment
Gradle 7.4

**Comment by wolfs** [comment]
Thank you for reporting! It seems like the `equals` in `PluginDeclaration` doesn't take `displayName` and `description` into account. I wonder why that is. In any case, the `GeneratePluginDescriptors` task probably shouldn't declare the list of plugin descriptors directly as an input.

**Comment by wolfs** [comment]
@DanielThomas This will be fixed in Gradle 7.5.

**Comment by hegyibalint** [comment]
Fixes #19029
